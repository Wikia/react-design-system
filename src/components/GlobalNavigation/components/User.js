import React from 'react';
import PropTypes from 'prop-types';

import Link from './Link';
import linkType from '../linkType';

const User = ({user}) => {
  if (!user) {
    return null;
  }

  return (
    <div className="wds-global-navigation__user-menu wds-has-shadow wds-dropdown">
      <div className="wds-dropdown__toggle">
        <div className="wds-avatar">
          <img
            src={user.avatar}
            alt={user.username}
            title={user.username}
            className="wds-avatar__image"
          />
        </div>
        <svg className="wds-icon wds-icon-tiny wds-dropdown__toggle-chevron">
          <use xlinkHref="#wds-icons-dropdown-tiny" />
        </svg>
      </div>
      <div className="wds-is-right-aligned wds-dropdown__content">
        <ul className="wds-list wds-is-linked">
          {user.items.map(item => (
            <li key={item.tracking_label}>
              <Link link={item} />
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

User.propTypes = {
  user: PropTypes.shape({
    username: PropTypes.string.isRequired,
    avatar: PropTypes.string,
    tracking_label: PropTypes.string,
    items: PropTypes.arrayOf(linkType),
  }),
};

User.defaultProps = {
  user: null,
};

export default User;
