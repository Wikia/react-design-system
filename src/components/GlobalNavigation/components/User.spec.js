import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import sinon from 'sinon';

import User from './User';

test('User shouldn\'t render with null', () => {
  const component = renderer.create(
    <User user={null} />
  );
  expect(component.toJSON()).toBe(null);
});

test('User should render correctly with data', () => {
  const user = {
    avatar: 'https://static.wikia.nocookie.net/2536a38e-ab79-4d85-a5a0-16428e2582e8/scale-to-width-down/50',
    username: 'User name',
    tracking_label: 'account',
    items: [
      {
        type: 'link-text',
        title: 'View wiki profile',
        href: 'http://community.wikia.com/wiki/User:User_name',
        tracking_label: 'account.profile-wiki',
      },
      {
        type: 'link-text',
        title: 'View author profile',
        href: 'http://fandom.wikia.com/u/User_name',
        tracking_label: 'account.profile',
      },
      {
        type: 'link-logout',
        title: 'Sign-out',
        href: 'https://www.sandbox-content.wikia.com/signout?redirect=<referrer>',
        tracking_label: 'account.sign-out',
      },
    ],
  };
  const component = renderer.create(
    <User user={user} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
