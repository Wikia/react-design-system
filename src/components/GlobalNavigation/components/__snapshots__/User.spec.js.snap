// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render correctly with data 1`] = `
<div
  className="wds-global-navigation__user-menu wds-has-shadow wds-dropdown"
>
  <div
    className="wds-dropdown__toggle"
  >
    <div
      className="wds-avatar"
    >
      <img
        alt="User name"
        className="wds-avatar__image"
        src="https://static.wikia.nocookie.net/2536a38e-ab79-4d85-a5a0-16428e2582e8/scale-to-width-down/50"
        title="User name"
      />
    </div>
    <svg
      className="wds-icon wds-icon-tiny wds-dropdown__toggle-chevron"
    >
      <use
        xlinkHref="#wds-icons-dropdown-tiny"
      />
    </svg>
  </div>
  <div
    className="wds-is-right-aligned wds-dropdown__content"
  >
    <ul
      className="wds-list wds-is-linked"
    >
      <li>
        <a
          data-tracking-label="account.profile-wiki"
          href="http://community.wikia.com/wiki/User:User_name"
        >
          View wiki profile
        </a>
      </li>
      <li>
        <a
          data-tracking-label="account.profile"
          href="http://fandom.wikia.com/u/User_name"
        >
          View author profile
        </a>
      </li>
      <li>
        <div
          className=""
        >
          <div
            id="global-navigation-user-sign-out"
          >
            <form
              action="https://www.sandbox-content.wikia.com/signout?redirect=<referrer>"
              method="POST"
            >
              <button
                className="wds-sign-out__button"
                data-tracking-label="account.sign-out"
                type="submit"
              >
                Sign-out
              </button>
            </form>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
